prediction_market_agent/agents/crewai_subsequential_agent/deploy.py

```diff
@@ -0,0 +1,71 @@
+import random
+from decimal import Decimal
+
+from prediction_market_agent_tooling.deploy.agent import DeployableAgent
+from prediction_market_agent_tooling.markets.agent_market import AgentMarket
+from prediction_market_agent_tooling.markets.data_models import BetAmount, Currency
+from prediction_market_agent_tooling.markets.markets import MarketType
+
+from prediction_market_agent.agents.crewai_subsequential_agent.crewai_agent_subquestions import (
+    CrewAIAgentSubquestions,
+)
+from prediction_market_agent.agents.utils import market_is_saturated
+
+
+class DeployableThinkThoroughlyAgent(DeployableAgent):
+    # For cheaper credits at this experimental stage
+    def __init__(self) -> None:
+        super().__init__()
+
+    def pick_markets(self, markets: list[AgentMarket]) -> list[AgentMarket]:
+        # We simply pick 5 random markets to bet on
+        picked_markets: list[AgentMarket] = []
+        random.shuffle(markets)
+        for market in markets:
+            # Assume very high probability markets are already known, and have
+            # been correctly bet on, and therefore the value of betting on them
+            # is low.
+            if not market_is_saturated(market=market):
+                picked_markets.append(market)
+                if len(picked_markets) == 5:
+                    break
+
+        return picked_markets
+
+    def answer_binary_market(self, market: AgentMarket) -> bool:
+        # The answer has already been determined in `pick_markets` so we just
+        # return it here.
+        result = CrewAIAgentSubquestions().answer_binary_market(market.question)
+        return True if result.decision == "y" else False
+
+    def calculate_bet_amount(self, answer: bool, market: AgentMarket) -> BetAmount:
+        if market.currency == Currency.xDai:
+            return BetAmount(amount=Decimal(0.1), currency=Currency.xDai)
+        else:
+            raise NotImplementedError("This agent only supports xDai markets")
```

Comment:
Is there any reason not to deploy it on Manifold?

Reply:
```suggestion
            return super(). calculate_bet_amount(answer, market)
```

This should be okay 🙏

Reply:
No, I copied this from KnownOutcomeAgent - I will default back to the base class implementation, great catch!